packages/design-tokens-schema/src/basis-tokens.ts

```diff
@@ -189,3 +223,34 @@ export const ThemeSchema = z
   });
 
 export type Theme = z.infer<typeof ThemeSchema>;
+
+export const addContrastExtensions = (rootConfig: Theme) => {
+  walkColors(rootConfig, (color, path) => {
+    const lastPath = path.at(-1)! as ColorNameKey;
+
+    // Check that we have listed this color to have a known contrast counterpart
+    if (!(lastPath in CONTRAST) || !CONTRAST[lastPath]) return;
+
+    // Make sure $extensions exists
+    color.$extensions ??= {};
```

Comment:
Never seen this one in the wild before 😅

Reply:
Eh, sonarqube demanded it